### PR TITLE
Docusaurus2

### PR DESCRIPTION
**Pull request by allanchau**
This PR migrates to Docusaurus 2.

**Setup**

- [x] cd into the `docusaurus` directory and run `yarn`.

**Notes**

- The index page looks a bit different due to the deprecated blocks package from v1. I haven't updated it to look 100% like the current docs site.

**Testing**

- [x] Run `yarn start` and go to http://localhost:3000/idearium-lib/.
- [x] Make sure the docs look correct.

**Deployment**

- [ ] ~~Run `GIT_USER=<GITHUB_USERNAME> yarn deploy`~~ This should be automatic now!

### CODE REVIEW

File: docusaurus/README.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+```console
+yarn build
+```
+
+This command generates static content into the `build` directory and can be served using any static contents hosting service.
+
+## Deployment
+
+```console
+GIT_USER=<Your GitHub username> USE_SSH=true yarn deploy
```
Comment (smebberson) [inline docusaurus/README.md:30]: @allanchau, can we improve this? We don't really use SSH for Git authentication and therefore requires some setup.

File: docusaurus/docs/encryption.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
 ---
 id: encryption
-title: @idearium/encryption
+title: '@idearium/encryption'
```
Comment (smebberson) [inline docusaurus/docs/encryption.md:3]: Does this need `sidebar_label` and `slug`?

File: docusaurus/docs/lists.md [hunk 3]
```diff
@@ -1,6 +1,6 @@
 ---
 id: lists
-title: @idearium/lists
+title: '@idearium/lists'
```
Comment (smebberson) [inline docusaurus/docs/lists.md:3]: Does this need `sidebar_label` and `slug`?

File: docusaurus/docs/log.md [hunk 4]
```diff
@@ -1,6 +1,6 @@
 ---
 id: log
-title: @idearium/log
+title: '@idearium/log'
```
Comment (smebberson) [inline docusaurus/docs/log.md:3]: Does this need `sidebar_label` and `slug`?

File: docusaurus/docusaurus.config.js [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    onBrokenLinks: 'throw',
+    onBrokenMarkdownLinks: 'warn',
+    organizationName: 'idearium',
+    presets: [
+        [
+            '@docusaurus/preset-classic',
+            {
+                docs: {
+                    editUrl:
+                        'https://github.com/idearium/idearium-lib/tree/master/docusaurus/',
```
Comment (smebberson) [inline docusaurus/docusaurus.config.js:15]: This is incorrect.

File: docusaurus/package.json [hunk 6]
```diff
@@ -0,0 +1,34 @@
+{
+    "name": "docusaurus",
+    "version": "0.0.0",
```
Comment (smebberson) [inline docusaurus/package.json:3]: Should we keep this at version `1.0.0`?

**Review by smebberson — CHANGES_REQUESTED** [review]
@allanchau, a couple of things.

File: docusaurus/docusaurus.config.js (hunk 5 above)
Comment (allanchau) [reply docusaurus/docusaurus.config.js:15]: @smebberson I put this url there for when the monorepo eventually gets merged into master, at that point it should be correct?

Comment (smebberson) [reply docusaurus/docusaurus.config.js:15]: Why don't we put in the `feature-monorepo` branch for now?

File: docusaurus/docs/encryption.md (hunk 2 above)
Comment (allanchau) [reply docusaurus/docs/encryption.md:3]: @smebberson Doesn't look like it, it only needs to be there on the first page to make the sidebar show up.

File: docusaurus/README.md (hunk 1 above)
Comment (allanchau) [reply docusaurus/README.md]: @smebberson I've added a workflow based on https://v2.docusaurus.io/docs/deployment/#triggering-deployment-with-github-actions, hopefully that works.

**Comment by smebberson** [comment]
@allanchau, the check test failed in the new docusaurus workflow... https://github.com/idearium/idearium-lib/pull/73/checks?check_run_id=1478323616

**Comment by allanchau** [comment]
@smebberson I've fixed that up now.

**Review by smebberson — APPROVED** [review]
@allanchau, looks great. Please launch it :)

**Pull request merged**